merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/ProfilingState.java

```diff
@@ -14,4 +14,8 @@ ProfilingState<DynamicsType> create(final Resource<DynamicsType> resource) {
   public void append(final Duration currentTime, final Querier querier) {
     this.profile.append(currentTime, this.resource.getDynamics(querier));
   }
+
+  public ProfilingState<?> duplicate() {
+    return new ProfilingState<>(resource, profile.duplicate());
+  }
```

Comment:
@mattdailis: Clarity: from my understanding, `resource` isn't duplicated because the value of a tracked resource is stored in `profile`, while `resource` is generally defining the shape of the resource, with `getDynamics` delegating to an underlying `CellRef`. Is this correct? If it is, could you clarify why Cells don't need to be duplicated (or if they do, where that happens)?

Reply:
Good question - the cell duplication logic actually already existed prior to this PR. It's used to branch cells, so that concurrently executing tasks can each get their own view into history at a particular time step. This is what happens when you call `new LiveCells` with an old LiveCells as a parameter - it will use the old LiveCells as a starting point, but will make copies of its cells before stepping them up